### PR TITLE
libspnav: init at 0.2.3

### PR DESCRIPTION
**Pull request by flosse**

### CODE REVIEW

**Comment by fpletz** [comment]
The expression for `libspnav-cube-example` isn't referenced in `all-packages.nix` and thus can't be used. As it's just an example, is it really necessary? In this case it should start with lib as it's an executable, IMHO.

**Comment by flosse** [comment]
> As it's just an example, is it really necessary?

it's at least very helpful to test if the 3D mouse works.

> In this case it should start with lib as it's an executable

but it starts wit lib!? Or what is the point? How exactly should it be named?

**Comment by fpletz** [comment]
> but it starts wit lib!? Or what is the point? How exactly should it be named?

Personally, I would call it `spnav-cube-example`, but that's just be my preference. Do what you think is best. :)

Okay, let's keep it, just add it to `all-packages.nix` and I would merge this. Thanks.

**Comment by jgillich** [comment]
Since the example is part of `libspnav`, why not add it to the main package and make it optional?

**Comment by flosse** [comment]
>  why not add it to the main package and make it optional?

I like that idea. I'll see if I can do that.

**Comment by fpletz** [comment]
:+1:

**Comment by flosse** [comment]
Would you mind to review the change? Is there a better way to do it?

**Comment by flosse** [comment]
FYI: FreeCAD fails on travis because it takes too much time to compile:

    The job exceeded the maxmimum time limit for jobs, and has been terminated

see [log](https://s3.amazonaws.com/archive.travis-ci.org/jobs/102812514/log.txt)

**Comment by globin** [comment]
I'd prefer splitting the cube example into a seperate derivation called spnav-cube-example referenced from all-packages.nix

**Comment by flosse** [comment]
@fpletz do you see any points that needs to be changed?

> I'd prefer splitting the cube example into a seperate derivation called spnav-cube-example referenced from all-packages.nix

That's what it was first but now I can't see any advantage anymore.  So are there any reasons why you want to maintain a separate derivation?

File: nixos/modules/services/hardware/spacenavd.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        default = false;
+        description = ''
+          Whether to enable spacenavd to support 3DConnexion devices
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.spacenavd = {
```
Comment (fpletz) [inline nixos/modules/services/hardware/spacenavd.nix]: Is it necessary to run this service as root?

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -7363,6 +7363,9 @@ let
 
   libspectre = callPackage ../development/libraries/libspectre { };
 
+  libspnav = callPackage ../development/libraries/libspnav { };
+  spacenavd = callPackage ../misc/drivers/spacenavd { };
```
Comment (fpletz) [inline pkgs/top-level/all-packages.nix]: `spacenavd` should be moved to the misc section (look for `### MISC`) in `all-packages.nix`, that should be around line 15300.

**Comment by fpletz** [comment]
> That's what it was first but now I can't see any advantage anymore. So are there any reasons why you want to maintain a separate derivation?

Well, it's not intuitive to have to add a library to the system path to get the executable to test the device. If another program, like `freecad` or `spacenavd` depends on `libspnav` and the user adds this to this nixos configuration, the executable `spnav-cube-example` won't be in the path and thus won't be available to the user. It will, however, be part of the `libspnav` derivation and present in the nix store.

You could split those two up but refer to the same `src` and `version`. You can for example look at 03959d46165f49e6c3629cf1bf2a9ce0f6933de3 where I did this for `whois` because it's part of the source tarball of `mkpasswd`.

Another option would be to use another output for `spnav-cube-example` or the lib itself. For instance, look at `xfsprogs`.

**Comment by flosse** [comment]
Thanks for your feedback! I'll work on it next week.

**Comment by rasendubi** [comment]
(triage) @flosse any update?

**Comment by flosse** [comment]
@rasendubi It was a back and forth so I lost a bit the motivation :-\ So feel free to to finish it ;-)

**Comment by joachifm** [comment]
Closing as abandoned.

**Pull request closed without merging**